### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by ugnet**

### CODE REVIEW

File: frontend/src/data/faculties.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+    { text: 'Komunikacijos fakultetas', id: 9 },
+    { text: 'Matematikos ir informatikos fakultetas', id: 10 },
+    { text: 'Medicinos fakultetas', id: 11 },
+    { text: 'Politikos mokslų institutas', id: 12 },
+    { text: 'Teisės fakultetas', id: 13 },
+    { text: 'Verslo mokykla', id: 14 },
 ];
 
-export default faculties;
+export default faculties;
```
Comment (PaperSoda) [inline frontend/src/data/faculties.js]: New line at the end of file

File: frontend/src/data/programs.js [hunk 2]
```diff
@@ -1,3 +1,114 @@
...
+        { text: "Leidyba ir reklama" },
+        { text: "Mokslinės informacijos valdymas" },
+        { text: "Verslo informacijos vadyba" },
+        { text: "Žinių ir inovacijų vadyba" },
+        { text: "Žurnalistika" }
+    ],
+    10: [{ text: "Programų sistemos" },
+        { text: "Bioinformatika" },
+        { text: "Ekonometrija" },
+        { text: "uomenų mokslas" },
```
Comment (PaperSoda) [inline frontend/src/data/programs.js]: Typo

File: frontend/src/store/modules/auth.module.js [hunk 3]
```diff
@@ -1,80 +1,80 @@
...
 // import { LOGIN, REGISTER, LOGOUT } from '../types/action.type';
 import api from '../../api/index';
 
 const state = {
-  currentUser: {},
-  isAuthenticated: false,
-  errors: [],
-  registerFailed: false,
+    currentUser: {},
+    isAuthenticated: false,
```
Comment (PaperSoda) [inline frontend/src/store/modules/auth.module.js]: tau rodo editorius tuos space pakeist? nes ant manes amzinai pyksta del spacingo ir nori mazesniu

File: frontend/src/store/modules/auth.module.js [hunk 4]
```diff
@@ -1,80 +1,80 @@
...
-  state,
-  actions,
-  getters,
-  mutations,
-};
+    state,
+    actions,
+    getters,
+    mutations,
+};
```
Comment (PaperSoda) [inline frontend/src/store/modules/auth.module.js]: new line at the end of file

File: frontend/src/views/Register.vue [hunk 5]
```diff
@@ -1,15 +1,145 @@
...
+  data() {
+    return {
+      username: '',
+      password: '',
+      email: '',
+      selectedFaculty: null,
+      selectedStudyProgram: null,
+      semester: "",
+      options: [{ value: null, text: 'Please select an option', id: 0 }],
+      optionssp: [{ value: null, text: 'Please select an option' }],
```
Comment (PaperSoda) [inline frontend/src/views/Register.vue]: bad name

File: frontend/vue.config.js [hunk 6]
```diff
@@ -1,7 +1,8 @@
...
-    proxy: 'https://localhost:44301',
-  },
-};
+    // The URL where the .Net Core app will be listening.
+    // Specific URL depends on whether IISExpress/Kestrel and HTTP/HTTPS are used
+    devServer: {
+        proxy: 'https://localhost:44301',
+    },
+    lintOnSave: false
+};
```
Comment (PaperSoda) [inline frontend/vue.config.js]: new line at the end of file

File: frontend/src/views/Register.vue [hunk 7]
```diff
@@ -1,15 +1,145 @@
...
+    flex-direction: column;
+    height: min-content;
+    padding: 40px;
+    background-color: #eee;
+    border-radius: 4px;
+  }
+  .small-text {
+    font-size: 80%;
+  }
+</style>
```
Comment (PaperSoda) [inline frontend/src/views/Register.vue]: new line at the end of file

File: frontend/src/views/Register.vue [hunk 8]
```diff
@@ -1,15 +1,145 @@
...
+      value: faculty,
+      text: faculty.text,
+      id: faculty.id
+    }))];
+  },  
+  methods: {
+    ...mapActions([REGISTER]),
+    onSubmit() {
+      this.register({ username: this.username, password: this.password,
+      email: this.email, faculty: this.selectedFaculty.text, studyProgram: this.selectedStudyProgram.text,
```
Comment (PaperSoda) [inline frontend/src/views/Register.vue]: needs better formatting

File: frontend/src/views/Register.vue [hunk 9]
```diff
@@ -1,15 +1,145 @@
...
+      </b-form-group>
+
+      <b-form-group
+        id="input-group-7"
+        label="Semester"
+        label-for="input-7"
+        >
+        <select v-model="semester" id="inputState2" class="form-control">
+          <option value="1">1</option><option value="2">2</option> <option value="3">3</option>
+          <option value="4">4</option><option value="5">5</option> <option value="6">6</option>
```
Comment (PaperSoda) [inline frontend/src/views/Register.vue]: Needs better formatting

File: frontend/src/views/Register.vue [hunk 10]
```diff
@@ -1,15 +1,145 @@
...
+  methods: {
+    ...mapActions([REGISTER]),
+    onSubmit() {
+      this.register({ username: this.username, password: this.password,
+      email: this.email, faculty: this.selectedFaculty.text, studyProgram: this.selectedStudyProgram.text,
+      semester: this.semester});
+    },
+    loadPrograms(){
+      this.optionssp = [{ value: null, text: 'Please select an option' }];
+      this.optionssp = [...this.optionssp, ...studyPrograms[this.selectedFaculty.id].map(programe => ({
```
Comment (PaperSoda) [inline frontend/src/views/Register.vue]: this.optionssp = [{ value,... }, ...studyPrograms[this....]]

File: frontend/src/store/modules/auth.module.js (hunk 3 above)
Comment (ugnet) [reply frontend/src/store/modules/auth.module.js]: as kai issaugau man jie visi automatiskai pasidaro

File: frontend/src/views/Register.vue (hunk 10 above)
Comment (ugnet) [reply frontend/src/views/Register.vue]: cia jeigu be tos pirmos eilutes kai pasirenki vis nauja fakulteta, studiju programose susikaupia visu senesniu fakultetu studiju programos. tikriausiai yra geresnis budas kaip sita ispresti, bet nesugalvojau

File: backend/Program.cs [hunk 11]
```diff
@@ -14,6 +14,6 @@ public static void Main(string[] args)
 
         public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
-                .UseStartup<Startup>();
+                .UseStartup<Startup>().UseUrls("https://localhost:44301/");
```
Comment (PaperSoda) [inline backend/Program.cs]: would look better on new line:
  something
     .a()
     .b();

File: frontend/src/data/programs.js [hunk 12]
```diff
@@ -1,3 +1,114 @@
...
+        { text: "Verslumas" }
+    ],
+    14: [{ text: "Tarptautinis verslas" },
+        { text: "Tarptautinis verslas (anglų k.)" },
+        { text: "Tarptautinio verslo valdymas" },
+        { text: "Tarptautinė rinkodara" },
+        { text: "Verslo finansai" }
+    ]
+};
+
```
Comment (campingtree) [inline frontend/src/data/programs.js]: Don't really get why you guys rewriting this to a js object. We had a perfectly working JSON file.

**Comment by campingtree** [comment]
As it stands right now, there's a serious lack of communication between what API returns and what the user sees.
E.g. (register action only checks if registering was successful, if it was not, user has no way of knowing).:
```js
if (res.status !== 200) {
  commit({ SET_REGISTER });
}
```
Additionally, part of input validation could be done at frontend level, e.g. 
1. We shouldn't even let user submit, if for say email doesn't match a pattern and then if it does, we validate it again, just this time at the backend level.
2. Frontend should check for input length.
3. etc.....

**Not sure if it's the way it is because we don't have proper authentication yet, but something to keep in mind.**

**Pull request merged**